lib/directory-monitor.c

```diff
@@ -172,7 +172,12 @@ _collect_all_files(DirectoryMonitor *self, GDir *directory)
       gchar *filename_real_path = resolve_to_absolute_path(filename, self->real_path);
       event.full_path = build_filename(self->real_path, filename);
       event.event_type = g_file_test(filename_real_path, G_FILE_TEST_IS_DIR) ? DIRECTORY_CREATED : FILE_CREATED;
-      self->callback(&event, self->callback_data);
+
+      if(self->callback)
```

Comment:
I may revert that change since it used to be ``self->callback(&event, self->callback_data);`` and that check wasn't done anyway. (This seems outside of the scope for this PR).

Reply:
I think revert would be a best option; maybe an assert. But I would not do the `msg_error`, as that could be ignored.